sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs

```diff
@@ -331,13 +331,25 @@ public async Task DeleteAsync_ConditionsFail()
             }
         }
 
+        // It is possible for the service to have responded that the container was already created after
+        // creating the new FileSystem only to have the directory creation fail. This appears to be due to
+        // a propagation delay from the blob endpoint to the dfs endpoint, since the directory creation happens against
+        // the dfs endpoint.
+        // This method will create the directory and retry if a 404 is returned that indicates the filesystem doesn't exist.
+        private async Task<DataLakeDirectoryClient> CreateDirectoryEnsureFileSystemPropagated(DisposingFileSystem fileSystem, string directoryName)
+        {
+            return await RetryAsync(
+                async () => await fileSystem.FileSystem.CreateDirectoryAsync(directoryName),
+                ex => ex.Status == 404);
```

Comment:
Is there a more specific `ex.ErrorCode` we could check so it's not any 404?

Reply:
Yes, we can use FilesystemNotFound